host-spec/c04-networking.tm

```diff
@@ -81,551 +128,427 @@
+    body (optional)>|<cell|Def. <reference|sect-block-body>>>|<row|<cell|bytes>|<cell|>|<cell|>|<cell|>>|<row|<cell|bytes>|<cell|4>|<cell|Block
+    receipt (optional)>|<cell|>>|<row|<cell|bytes>|<cell|5>|<cell|Block
+    message queue (optional)>|<cell|>>|<row|<cell|bytes>|<cell|6>|<cell|Justification
+    (optional)>|<cell|Def. <reference|defn-grandpa-justification>>>|<row|<cell|bool>|<cell|7>|<cell|Indicates
+    whether the justification>|<cell|>>|<row|<cell|>|<cell|>|<cell|is empty
+    (i.e. should be ignored).>|<cell|>>>>>>
+      <strong|BlockData> Protobuf structure.
+    </big-table>
+  </definition>
+
+  <subsubsection|Transactions><label|sect-msg-transactions>
+
+  Transactions are sent directly in its full form to connected peers. It's
+  considered good behavior to implement a mechanism which only sends a
+  transaction once to each peer and avoids sending duplicates. Such a
+  mechanism is implementation specific and any absence of such a mechanism
+  can result in consequences which are undefined. The Polkadot Host much be
+  able to handle duplicate transactions received from its peers in order to
+  prevent the inclusion of duplicate transactions in multiple blocks.
+
+  The transactions message is represented by <math|M<rsub|T>> and is defined
+  as follows:
+
+  <\equation*>
+    M<rsub|T>\<assign\>Enc<rsub|SC><around*|(|C<rsub|1>,\<ldots\>,C<rsub|n>|)>
+  </equation*>
+
+  in which:
+
+  <\equation*>
+    C<rsub|i>\<assign\>Enc<rsub|SC><around*|(|E<rsub|i>|)>
+  </equation*>
+
+  Where each <math|E<rsub|i>> is a byte array and represents a sepearate
+  extrinsic. The Polkadot Host is indifferent about the content of an
+  extrinsic and treats it as a blob of data.
+
+  The exchange of transactions is conducted on the
+  <verbatim|/dot/transactions/1> substream.
+
+  <subsubsection|Grandpa Votes><label|sect-msg-grandpa>
+
+  The exchange of GRANDPA messages is conducted on the
+  <verbatim|/paritytech/grandpa/1> substream. The process for the creation of
+  such votes is described in Section <reference|sect-finality>.
+
+  <subsection|I'm Online Heartbeat>
+
+  The I'm Online heartbeat is a crucial part of the Polkadot validation
+  process, as it signals the active participation of validators and confirms
+  their reachability. The Polkadot network punishes unreachable validators
+  which have been elected to an authority by slashing their bonded funds.
+  This is achieved by requiring validators to issue an I'm Online heartbeat,
+  which comes in the form of a signed extrinsic, on the start of every Era.
+
+  The Polkadot Runtime fully manages the creation and the timing of that
+  signed extrinsic, but it's the responsiblity of the Host to gossip that
+  extrinsic to the rest of the network. When the Runtime decides to create
+  the heartbeat, it will call the <verbatim|ext_offchain_submit_transaction>
+  Host API as described in Section <reference|sect-ext-offchain-submit-transaction>.
```

Comment:
I'm not sure that this is worth mentioning. These heartbeat messages could theoretically disappear or change their nature at any runtime upgrade.